zaza/model.py

```diff
@@ -510,6 +510,10 @@ def check_unit_workload_status_message(model, unit, message=None,
         states = {}
     async with run_in_model(model_name) as model:
         check_model_for_hard_errors(model)
+        logging.info("Waiting for a unit to appear")
+        await model.block_until(
+            lambda: len(model.units) > 0
+        )
```

Comment:
I guess timeout=timeout is left out here because it is less likely that this step fails.